### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by scottcreynolds**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,21 +10,338 @@ JavaScript XHR
...
 
-## Testing?
+![smarf](http://i.giphy.com/fdyPkHljnYdEI.gif)
 
-This could be a tested code-along. For testing, stub out the calls to the API and use mock data.
+When we go to a library to research something (surely somebody still
+does this), we only request the books we need as we need them, and until
+we do, they remain stored safely on thier shelves out of everyone's way.
+
+Similarly, when working with server data, we often want to just request
```
Comment (pletcher) [inline README.md:36]: I 💟 this opening a ton

File: README.md [hunk 2]
```diff
@@ -10,21 +10,338 @@ JavaScript XHR
...
+  <a href="#" onclick="getRepositories()">Get Repositories</a>
+</div>
+```
+
+Then let's create our `getRepositories` function and initiate our XHR
+request.
+
+```js
+function getRepositories() {
+  var req = new XMLHttpRequest()
```
Comment (pletcher) [inline README.md]: Could we use `const` here?

File: README.md [hunk 3]
```diff
@@ -10,21 +10,338 @@ JavaScript XHR
...
+</div>
+```
+
+Then let's start by simply listing the repository names.
+
+```js
+function showRepositories(event, data) {
+  console.log(this.responseText)
+  var repoList = "<ul>"
+  for(var i=0;i < this.responseText.length; i++) {
```
Comment (pletcher) [inline README.md:196]: Students will know about `Array.prototype.map` at this point -- could we use that?

File: README.md [hunk 4]
```diff
@@ -10,21 +10,338 @@ JavaScript XHR
...
+
+Now if we reload and click our link...
+
+![jim hides](http://i.giphy.com/FGTVmzksb2j0k.gif)
+
+Okay not quite what we expected. While it might be fun to have a list of
+a million `undefined` values on a page, we got repositories to print
+out. What happened?
+
+The key lies in the `responseText` property. We can look at it and
```
Comment (pletcher) [inline README.md:212]: This is excellent!

File: README.md [hunk 5]
```diff
@@ -10,21 +10,338 @@ JavaScript XHR
...
+request when that link is clicked, and then show the commits in the
+second column.
+
+We'll start by adding the link to our repository output.
+
+```js
+function showRepositories(event, data) {
+  var repos = JSON.parse(this.responseText)
+  console.log(repos)
+  var repoList = "<ul>"
```
Comment (pletcher) [inline README.md]: e.g.,

``` js
const repoList = `<ul>${repos.map(r => '<li>' + r.name + ' - <a href="#" data-repo="' + name + '" onclick="getCommits(this)">Get Commits</a></li>').join('')}</ul>`
```

Comment (pletcher) [inline README.md]: Actually, we should use a handlebars template!

Comment (scottcreynolds) [inline README.md]: one of the next lessons in this unit is about using handlebars templates with this, so these would basically be duplicates of each other.

i think I might be a little confused about just how modular these are supposed to be

Comment (pletcher) [inline README.md]: Ah, okay -- we can keep this one like this, then :)

Modularity with this stuff is, I think, desirable but essentially impossible. We can always tweak them :)

File: README.md (hunk 3 above)
Comment (scottcreynolds) [inline README.md:196]: can't do it here because `responseText` doesn't `map` but can def do it in the later places after we parse JSON.

**Comment by scottcreynolds** [comment]
@pletcher updates made

**Comment by pletcher** [comment]
LGTM!

File: README.md (hunk 3 above)
Comment (pletcher) [inline README.md:196]: Ah, derp -- thanks for catching that :)

**Pull request merged**